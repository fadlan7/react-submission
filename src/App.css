.App {
  height: 900px;
  margin: 50px;
}
