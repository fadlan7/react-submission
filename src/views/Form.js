import React from "react";

class Form extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      makanan: [],
    };

    this.onPick = this.onPick.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.inputName = React.createRef();
    this.inputPekerjaan = React.createRef();
    this.inputAlamat = React.createRef();
    this.inputMakanan = React.createRef();
    this.inputTanggungan = React.createRef();
    this.inputCheck = React.createRef();
    this.inputFile = React.createRef();
    this.inputGender = React.createRef();
  }

  onPick(event) {
    let makanan = [...this.state.makanan];
    let idx = makanan.findIndex((elemen) => event.target.value === elemen);
    if (idx > 0) {
      makanan = [
        ...makanan.slice(0, idx),
        ...makanan.slice(idx + 1, makanan.length),
      ];
    } else if (idx === 0) {
      makanan = [...makanan.slice(idx + 1, makanan.length)];
    } else {
      makanan.push(event.target.value);
    }
    console.log(makanan);
    this.setState({ makanan });
  }

  onSubmit(event) {
    event.preventDefault();
    event.stopPropagation();
    alert(
      `Namaku adalah ${this.inputName.current.value}, pekerjaanku sebagai ${this.inputPekerjaan.current.value}, rumahku di ${this.inputAlamat.current.value}`
    );
  }

  render() {
    return (
      <div className="container">
        <h1>Form</h1>

        <form onSubmit={this.onSubmit}>
          <label>
            Name:
            <input name="name" type="text" ref={this.inputName} />
          </label>
          <br />
          <label>
            Pekerjaan:
            <input name="pekerjaan" type="text" ref={this.inputPekerjaan} />
          </label>
          <br />
          <label>
            Jenis Kelamin:
            <br />
            <label>
              Jenis Kelamin:
              <br />
              <input
                type="radio"
                name="gender"
                value="male"
                ref={this.inputGender}
              />
              <label>Male</label>
              <br />
              <input
                type="radio"
                name="gender"
                value="female"
                ref={this.inputGender}
              />
              <label>Female</label>
            </label>
          </label>
          <br />
          <label>
            Alamat:
            <textarea name="alamat" ref={this.inputAlamat} />
          </label>
          <br />
          <label>
            Makanan:
            <select multiple ref={this.inputMakanan} onChange={this.onPick}>
              <option value="pizza">Pizza</option>
              <option value="burger">Burger</option>
              <option value="cilok">Cilok</option>
            </select>
          </label>
          <br />
          <label>
            Tanggungan:
            <input name="tanggungan" type="number" ref={this.inputTanggungan} />
          </label>
          <br />
          <label>
            File:
            <input name="file" type="file" ref={this.inputFile} />
          </label>
          <br />
          <label>
            Apakah anda MAUU?
            <input name="checkbox" type="checkbox" ref={this.inputCheck} />
          </label>
          <br />
          <button type="submit">SUBMIT</button>
        </form>
      </div>
    );
  }
}

export default Form;
