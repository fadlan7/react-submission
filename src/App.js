import React from "react";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import { Home, Form, Users, ListKeys, LiftingState, Redux, ComposingExtracting } from "./views";
import "./App.css";
import "bootstrap/dist/css/bootstrap.min.css";
import { Navbar, Nav  } from "react-bootstrap";
// import {NavLink} from "react-bootstrap/NavLink";

function App() {
  return (
    <>
      <Router>
        <div>
          <header>
            <Navbar bg="dark" variant="dark" expand="lg">
              <Navbar.Brand href="#home">LoGo</Navbar.Brand>
              <Navbar.Toggle aria-controls="basic-navbar-nav" />
              <Navbar.Collapse id="basic-navbar-nav">
                <Nav className="mr-auto">
                  <Nav.Link>
                    <Link className="App-link" style={{textDecoration: 'none'}} to="/">
                      Home
                    </Link>
                  </Nav.Link>
                  <Nav.Link>
                    <Link className="App-link" style={{textDecoration: 'none'}} to="/form">
                      Form
                    </Link>
                  </Nav.Link>
                  <Nav.Link>
                    <Link className="App-link" style={{textDecoration: 'none'}} to="/router-dom">
                      Router-dom
                    </Link>
                  </Nav.Link>
                  <Nav.Link>
                    <Link className="App-link" style={{textDecoration: 'none'}} to="/list-keys">
                      List-keys
                    </Link>
                  </Nav.Link>
                  <Nav.Link>
                    <Link className="App-link" style={{textDecoration: 'none'}} to="/lifting-state">
                      Lifting-State
                    </Link>
                  </Nav.Link>
                  <Nav.Link>
                    <Link className="App-link" style={{textDecoration: 'none'}} to="/redux">
                      Redux
                    </Link>
                  </Nav.Link>
                  <Nav.Link>
                    <Link className="App-link" style={{textDecoration: 'none'}} to="/composing-extracting">
                      Composing-Extracting
                    </Link>
                  </Nav.Link>
                </Nav>
              </Navbar.Collapse>
            </Navbar>
          </header>

          <div className="App">
            <Switch>
              <Route exact path="/">
                <Home />
              </Route>
              <Route path="/form">
                <Form />
              </Route>
              <Route path="/router-dom">
                <Users />
              </Route>
              <Route path="/list-keys">
                <ListKeys />
              </Route>
              <Route path="/lifting-state">
                <LiftingState />
              </Route>
              <Route path="/redux">
                <Redux />
              </Route>
              <Route path="/composing-extracting">
                <ComposingExtracting />
              </Route>
            </Switch>
          </div>
        </div>
      </Router>

      <footer className='footer mt-auto py-3 bg-dark text-white'>
        <div className='container'>Footer</div>
      </footer>
    </>
  );
}

export default App;
