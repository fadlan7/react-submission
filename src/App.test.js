import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "./App";
import Form from "./views/Form";
import Redux from "./views/redux";
import LiftingState from "./views/lifting-state";
import Home from "./views/Home";
import ListKeys from './views/list-key';

// test("renders learn react", () => {
//   const { getByText } = render(<App />);
//   const linkElement = getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });

test("renders learn react", () => {
  const { getByText } = render(<Home />);
  const linkElement = getByText(/Home/i);
  expect(linkElement).toBeInTheDocument();
});

test("object equality", () => {
  const iniObject = render(<ListKeys/>);
  iniObject["kedua"] = 2;
  expect(iniObject).toEqual({
    pertama: 1,
    kedua: 2,
  });
});
