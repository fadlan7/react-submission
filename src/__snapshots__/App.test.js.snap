// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change language 1`] = `
<div
  className="container"
>
  <h1>
    Form
  </h1>
  <form
    onSubmit={[Function]}
  >
    <label>
      Name:
      <input
        name="name"
        type="text"
      />
    </label>
    <br />
    <label>
      Pekerjaan:
      <input
        name="pekerjaan"
        type="text"
      />
    </label>
    <br />
    <label>
      Jenis Kelamin:
      <br />
      <label>
        Jenis Kelamin:
        <br />
        <input
          name="gender"
          type="radio"
          value="male"
        />
        <label>
          Male
        </label>
        <br />
        <input
          name="gender"
          type="radio"
          value="female"
        />
        <label>
          Female
        </label>
      </label>
    </label>
    <br />
    <label>
      Alamat:
      <textarea
        name="alamat"
      />
    </label>
    <br />
    <label>
      Makanan:
      <select
        multiple={true}
        onChange={[Function]}
      >
        <option
          value="pizza"
        >
          Pizza
        </option>
        <option
          value="burger"
        >
          Burger
        </option>
        <option
          value="cilok"
        >
          Cilok
        </option>
      </select>
    </label>
    <br />
    <label>
      Tanggungan:
      <input
        name="tanggungan"
        type="number"
      />
    </label>
    <br />
    <label>
      File:
      <input
        name="file"
        type="file"
      />
    </label>
    <br />
    <label>
      Apakah anda MAUU?
      <input
        name="checkbox"
        type="checkbox"
      />
    </label>
    <br />
    <button
      type="submit"
    >
      SUBMIT
    </button>
  </form>
</div>
`;
